--- connection.hpp	2019-03-07 16:10:27.592472400 +0300
+++ connection.hpp.fix	2019-03-07 16:06:49.978481700 +0300
@@ -1483,6 +1483,8 @@
 // connection_factory_t
 //
 
+static std::atomic<std::uint64_t> m_connection_id_counter{ 1 };
+
 //! Factory for connections.
 template < typename Traits >
 class connection_factory_t
@@ -1532,7 +1534,7 @@
 		}
 
 	private:
-		std::uint64_t m_connection_id_counter{ 1 };
+		// std::uint64_t m_connection_id_counter{ 1 };
 
 		connection_settings_handle_t< Traits > m_connection_settings;
 
@@ -1543,4 +1545,4 @@
 
 } /* namespace impl */
 
-} /* namespace restinio */
\ No newline at end of file
+} /* namespace restinio */
